### ISSUE TITLE
How to Rubocop: CodeClimate, Travis or ...?

### ISSUE THREAD

**Issue by F3PiX** [rails-girls-summer-of-code/rgsoc-teams#858]
Dev happiness!

As of #857, we have Rubocop for static code analysis. 🎁 

We can run the analysis locally, with `bundle exec rubocop`, to solve issues before pushing a PR. 
 
With the current implementation, the analysis is part of the CI build. That means that the build will fail if there are Rubocop issues ( is this correct, @klappradla?) 

Another option is using CodeClimate. It has a very easy and beginner-friendly UI. Both for explaining issues and solving them. 

We already use CodeClimate. [Adding Rubocop is easy](https://docs.codeclimate.com/docs/rubocop):  add a codeclimate.yml,  enable rubocop, and then CC uses our rubocop.yml to enforce 🚔  the rules.
Then connect it to the GitHub PR's. 

With CC set up,  we would have 2 checks after submitting/updating a PR: one for the Travis build, and one for CodeClimate.  Which makes it superclear if there are either failing specs, or formatting/style issues. 

![jrducdgrgwghdg6zqmyw_github1](https://user-images.githubusercontent.com/6314687/30536408-fbe47104-9c65-11e7-9d9a-84405d5e897a.png)

Another option could be Travis build stages, see https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/857#discussion_r139322795 
I think that still misses the nice UI that CC offers. 

I'd vote for using CodeClimate, mostly because of the beginner-friendly UI plus the separate process in the PR workflow.  

Votes and thoughts please!

**Comment by carpodaster** [comment]
I'm also very much in favor of using CC for Rubocop.

**Comment by klappradla** [comment]
> That means that the build will fail if there are Rubocop issues

Yes, that's correct.

Some questions on the Code Climate thing (also related to ☝️ ): I don't understand the point of shelling out to a different service for two of the 4 steps in our CI pipeline 🙈  How can Code Climate prevent Travis from deploying to Heroku if a build does not comply to Rubocop or JSHint?

It's true, that things would be more readable if Travis would add a separate link on Github for each step - that's probably a reason why CircleCI is so popular nowadays 😉  - but I would say things are still reasonably easy to find there.

So tl;dr: I don't understand the benefits we get from using Code Climate and fear that it just adds a huge layer of complexity to the otherwise super basic and straight forward CI pipeline.
Maybe I just don't understand the tool yet.

**Comment by F3PiX** [comment]
CodeClimate is a different way of using Rubocop. It will do a separate check, and we set it up so that you can't _merge_ a PR without CC's ✅ . 

Would you mind giving it a try and evaluate in a month or so?

**Comment by klappradla** [comment]
Sure, go for it!

I may also on the side give Travis' pipelines a shot - it's something I'm really interested for further usage in other projects 😉

**Comment by F3PiX** [comment]
👍  I think it is better to have them both, because you and Carsten and some other do not always go through a PR !
